src/services/poolTogether.js

```diff
@@ -0,0 +1,416 @@
+  const currentDraw = await contract.getDraw(currentOpenDrawId);
+  const committedSupply = await contract.committedSupply();
+  const openSupply = await contract.openSupply();
+
+  let userInfo;
+  try {
+    const totalBalance = await contract.totalBalanceOf(address); // balance including open, committed, fees, sponsorship
+    if (!totalBalance.eq(0)) {
+      const openBalance = await contract.openBalanceOf(address); // balance in the open Draw
+      const userCurrentPoolBalance = await contract.balanceOf(address); // balance in the current committed Draw
+      const ticketBalance = openBalance.add(userCurrentPoolBalance); // this is the current ticket balance in total
+      userInfo = {
+        openBalance: formatMoney(utils.formatUnits(openBalance.toString(), unitType)),
+        totalBalance: formatMoney(utils.formatUnits(totalBalance.toString(), unitType)),
+        userCurrentPoolBalance: formatMoney(utils.formatUnits(userCurrentPoolBalance.toString(), unitType)),
+        ticketBalance: formatMoney(utils.formatUnits(ticketBalance.toString(), unitType)),
+      };
+    }
+  } catch (e) {
+    reportLog('Error checking PoolTogether User Info', {
+      address,
+      contractAddress,
+      symbol,
+      message: e.message,
+    }, Sentry.Severity.Error);
+  }
+
+  const supplyRatePerBlock = await contract.supplyRatePerBlock();
+
+  let currentPrize = ptUtils.toBN(0);
+  let prizeEstimate = ptUtils.toBN(0);
+  let drawDate = 0;
+  let remainingTimeMs = 0;
+  let totalPoolTicketsCount = 0;
+  if (balance && openSupply && committedSupply && accountedBalance && currentDraw && supplyRatePerBlock) {
+    const totalSupply = openSupply.add(committedSupply);
+    totalPoolTicketsCount = parseInt(utils.formatUnits(totalSupply.toString(), unitType), 10);
+    const { feeFraction, openedBlock } = currentDraw;
+
+    const prizeIntervalMs = symbol === DAI ? 604800000 : 86400000; // DAI weekly, USDC daily
+    const { timestamp: blockTimestamp } = await provider.getBlock(openedBlock.toNumber());
+    drawDate = (blockTimestamp.toString() * 1000) + prizeIntervalMs; // adds 14 days to a timestamp in miliseconds
+
+    remainingTimeMs = drawDate - balanceTakenAt.getTime();
+    remainingTimeMs = remainingTimeMs < 0 ? 0 : remainingTimeMs;
+    const remainingTimeS = remainingTimeMs > 0 ? remainingTimeMs / 1000 : prizeIntervalMs / 1000;
+    const remainingBlocks = remainingTimeS / 15.0; // about 15 second block periods
+    const blockFixedPoint18 = utils.parseEther(remainingBlocks.toString());
+
+    const prizeSupplyRate = ptUtils.calculatePrizeSupplyRate(
+      supplyRatePerBlock,
+      feeFraction,
+    );
+
+    currentPrize = ptUtils.calculatePrize(
+      balance,
+      accountedBalance,
+      feeFraction,
+    );
+    prizeEstimate = ptUtils.calculatePrizeEstimate(
+      balance,
+      currentPrize,
+      blockFixedPoint18,
+      prizeSupplyRate,
+    );
+  }
+
+  return Promise.resolve({
```

Comment:
If the function is `async`, `return`ing will automatically `Promise.resolve`.

Reply:
Just a note - this is how things work in Node but unsure in this environment.

Reply:
Proposed changes work good 👍